src/components/train/train-wagon/train-wagon.ts

```diff
@@ -204,31 +184,11 @@ export class SbbTrainWagonElement extends SlotChildObserver(LitElement) {
           ? html`<span class="sbb-screenreaderonly">, ${this.additionalAccessibilityText}</span>`
           : nothing}
         ${this.type === 'wagon'
-          ? html`<span
-              class="sbb-train-wagon__icons"
-              ?hidden=${!this._icons || this._icons?.length === 0}
-            >
-              ${this._icons?.length > 1
-                ? html`<ul
-                    class="sbb-train-wagon__icons-list"
-                    aria-label=${i18nAdditionalWagonInformationHeading[this._language.current]}
-                  >
-                    ${this._icons.map(
-                      (_, index) =>
-                        html`<li class="sbb-train-wagon__icons-item">
-                          <slot name=${`sbb-train-wagon-icon-${index}`}></slot>
-                        </li>`,
-                    )}
-                  </ul>`
-                : nothing}
-              <span class="sbb-train-wagon__icons-item" ?hidden=${this._icons?.length !== 1}>
-                ${this._icons?.length === 1
-                  ? html`<span class="sbb-screenreaderonly">
-                      ${i18nAdditionalWagonInformationHeading[this._language.current]}
-                    </span>`
-                  : nothing}
-                <slot></slot>
-              </span>
+          ? html`<span class="sbb-train-wagon__icons" ?hidden=${this.listChildren.length === 0}>
+              ${this.renderList({
+                class: 'sbb-train-wagon__icons-list',
+                ariaLabel: i18nAdditionalWagonInformationHeading[this._language.current],
```

Comment:
I think a "presentation" element can't have an ariaLabel, this is why before it was added in a span. This could be added into the mixin. for the labelledby casees it's more difficult. But I also see that there were a lot of cases where this was not handled I think it should be also before.